### PR TITLE
show all modules in history in modulegroup active group

### LINKED ISSUE #10343: [FR] modulegroups - add new "in history" group
I often find myself experimenting with a number of modules to get the result I want and I'm forever switching modules on and off as part of my experiments. I know that the "active" module group retains switched-off modules until it is next refreshed but I still find myself having to search or click through tabs to re-enable modules that I temporarily disabled a few steps ago.

I would therefore like to be able to replace the "active" module group with a new "in history" module group that displays all modules that are present in my history stack, regardless of whether or not they are currently enabled.

For maximum flexibility this should be offered "as well as" or "instead of" the active modules group, perhaps selectable in a combobox.

@AlicVB what do you think?

### PR DESCRIPTION
**Pull request by AlicVB**
this fix #10343 

- this add an option to modulegroups to show all modules present in history inside modulegroup active tab. the option can be activated in the preset manager or directly by right-clicking on the active icon
- this add ctrl-click to an entry of the history lib to show the corresponding module.

Please test. I hope that I don't break anything, esp. with the last commit which touch a sensible area...

### CODE REVIEW

**Comment by AlicVB** [comment]
@TurboGit : if possible, I would prefer that PR #10340 is merged first as it's more critical. Once that done, I'll have to take care of the rebasing here, as both PR touch same parts of the code... Thanks !

**Comment by elstoc** [comment]
I think this needs something to update the group if a new module is added to the stack. If I enter an image where, for example, sharpen is not present in the history stack, and then I enable it, the history module group is not updated.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Select "active group", go to pref and enable the "show all history in active group". Keep pref and see that the group has not been updated.

File: src/libs/modulegroups.c [hunk 1]
```diff
@@ -3763,6 +3823,11 @@ static void _manage_show_window(dt_lib_module_t *self)
   gtk_widget_set_name(d->basics_chkbox, "modulegroups_editor_setting");
   g_signal_connect(G_OBJECT(d->basics_chkbox), "toggled", G_CALLBACK(_manage_editor_basics_toggle), self);
   gtk_box_pack_start(GTK_BOX(vb), d->basics_chkbox, FALSE, TRUE, 0);
+  d->edit_full_active_cb = gtk_check_button_new_with_label(_("show all history in active group"));
```
Comment (TurboGit) [inline src/libs/modulegroups.c:3826]: I would add a tooltip:
"the active group will show enabled and disabled modules"

(BTW, the active group in this scenario is no more "active group" but more "history group")

Comment (TurboGit) [reply src/libs/modulegroups.c:3826]: Or "should modules in the history regardless of whether or not they are currently enabled"

**Comment by TurboGit** [comment]
BTW, you introduced the Ctrl+Click to show&expand the module in the right panel (fine - even if not terribly usable as you click on the left of the screen to then go on the right...) and furthermore there was a proposal from @elstoc about using the Ctrl+Click to add instance of the module on top of the history stack which I found more useful. Something to be thought about this...

**Comment by elstoc** [comment]
I'm not precious about what shortcuts are used for what feature in history stack, though perhaps Shift makes more sense with the less "intrusive" option (this one).

**Comment by AlicVB** [comment]
thanks @elstoc & @TurboGit for your reviews.
I should have fixed all your comments : 
- fix the refreshing issues
- add "show modules in the history regardless of whether or not they are currently enabled" as tooltip
- use shift-click in place of ctrl-click in history lib.

File: src/libs/modulegroups.c [hunk 2]
```diff
@@ -3825,8 +3840,11 @@ static void _manage_show_window(dt_lib_module_t *self)
   gtk_box_pack_start(GTK_BOX(vb), d->basics_chkbox, FALSE, TRUE, 0);
   d->edit_full_active_cb = gtk_check_button_new_with_label(_("show all history in active group"));
   gtk_widget_set_name(d->edit_full_active_cb, "modulegroups_editor_setting");
-  g_signal_connect(G_OBJECT(d->edit_full_active_cb), "toggled", G_CALLBACK(_manage_editor_full_active_toggle),
-                   self);
+  gtk_widget_set_tooltip_text(
+      d->edit_full_active_cb,
+      _("show modules in the history regardless of whether or not they are currently enabled"))
```
[the remark this answers is not in the document]
Comment (elstoc) [reply src/libs/modulegroups.c:3845]: Also this needs a semicolon at the end and the following line needs to be correctly indented. I had to fix that to make it compile.

**Comment by elstoc** [comment]
If I try to duplicate the "search only" preset with a new name and then tick the new "show all history" box in my duplicate version, the change is not saved and I can't select the duplicate preset.

File: src/libs/modulegroups.c [hunk 3]
```diff
@@ -2628,6 +2660,36 @@ static gboolean _manage_direct_basic_popup(GtkWidget *widget, GdkEventButton *ev
...
+
+static gboolean _manage_direct_active_popup(GtkWidget *widget, GdkEventButton *event, dt_lib_module_t *self)
+{
+  if(event->type == GDK_BUTTON_PRESS && event->button == 3)
+  {
+    dt_lib_modulegroups_t *d = (dt_lib_modulegroups_t *)self->data;
+    GtkWidget *pop = gtk_menu_new();
+    gtk_widget_set_name(pop, "modulegroups-popup");
+
+    GtkWidget *smt = gtk_check_menu_item_new_with_label(_("show all module in history"));
```
Comment (elstoc) [inline src/libs/modulegroups.c:2681]:
```suggestion
    GtkWidget *smt = gtk_check_menu_item_new_with_label(_("show history modules in active group"));
```

I'd like something shorter but I think that might be the shortest I can come up with at the moment

File: src/libs/modulegroups.c (hunk 1 above)
Comment (elstoc) [reply src/libs/modulegroups.c:3841]:
```suggestion
  d->edit_full_active_cb = gtk_check_button_new_with_label(_("show all history modules in active group"));
```

**Comment by elstoc** [comment]
Sorry I think I'm done adding comments now :)

File: src/libs/modulegroups.c (hunk 3 above)
Comment (elstoc) [reply src/libs/modulegroups.c:2681]: Same in the other instance of this checkbox too

Comment (AlicVB) [reply src/libs/modulegroups.c:2681]: hmm... are you sure the last part is really needed ? the item is only shown after a right click on the active group icon, so imho we can presume that the user know that the option will be apply there... (but I'm not against your suggestion either)

Comment (elstoc) [reply src/libs/modulegroups.c:2681]: Ah ok then that makes sense. So this checkbox should say "show all history modules" (and should have a tooltip the same as the other) and the presets window should have the label "show all history **modules** in active group"

**Comment by TurboGit** [comment]
@AlicVB : Does not compile on my side ?

```
/home/obry/dev/builds/darktable/x86_64-linux-gnu-default/src/src/libs/modulegroups.c: In function ‘_manage_show_window’:
/home/obry/dev/builds/darktable/x86_64-linux-gnu-default/src/src/libs/modulegroups.c:3867:120: error: expected ‘;’ before ‘g_signal_connect_data’
 3867 |       _("show modules that are present in the history stack, regardless of whether or not they are currently enabled"))
      |                                                                                                                        ^
      |                                                                                                                        ;
compilation terminated due to -Wfatal-errors.
```

**Comment by elstoc** [comment]
> Does not compile on my side

I mentioned that above - missing semicolon and messed up indenting

**Comment by TurboGit** [comment]
No problem anyway, the issue is indeed trivial. I have fixed it, tested and all ok for me so merged.

**Comment by TurboGit** [comment]
> I mentioned that above - missing semicolon and messed up indenting

For some reasons this was in a "resolved" collapsed block !!!

**Comment by AlicVB** [comment]
@TurboGit : thanks !
@elstoc : I may have been to quick to merge your proposal... and direct github commit, I have not tested it, sorry !
I'll still investigate your duplicate issue and fix it in another PR.

thanks again both for your reviews and tests

**Pull request merged**